app/enssubdomain/views.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+'''
+    Copyright (C) 2017 Gitcoin Core
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+'''
+
+import datetime
+
+from django.conf import settings
+from django.http import Http404, JsonResponse
+from django.shortcuts import redirect
+from django.template.response import TemplateResponse
+from django.utils import timezone
+from django.views.decorators.csrf import csrf_exempt
+
+from dashboard.views import w3
+from ens import ENS
+from eth_account.messages import defunct_hash_message
+from web3 import HTTPProvider, Web3
+
+from .models import ENSSubdomainRegistration
+
+ns = ENS.fromWeb3(w3)
+w3 = Web3(HTTPProvider(settings.WEB3_HTTP_PROVIDER))
+
+@csrf_exempt
+def ens_subdomain(request):
+    """Register ENS Subdomain."""
+    github_handle = request.user.username.lower()
+    if request.method == "POST":
+        signedMsg = request.POST.get('signedMsg', '')
+        signer = request.POST.get('singer', '').lower()
+        if signedMsg and signer and github_handle:
+            message_hash = defunct_hash_message(text="Github Username : {}".format(github_handle))
+            recovered_signer = w3.eth.account.recoverHash(message_hash, signature=signedMsg).lower()
+            if recovered_signer == signer:
+                txn_hash = ns.setup_owner("{}.{}".format(github_handle, settings.ENS_TLD), signer)
```

Comment:
@owocki This is the actual call that needs to be tested.

Reply:
got it; and we're blocked from testing because no one has a local geth node, right?  did you ever hear back about whether or not we can use infura?

Reply:
Since python ENS package was written some time ago, it only [works with keys hosted by your node](https://gitter.im/ethereum/web3.py?at=5ad4e4a55f188ccc156f7d10).

Reply:
https://github.com/gitcoinco/web/issues/901